hosted/eks/p1/p1_suite_test.go

```diff
@@ -94,3 +100,145 @@ func updateClusterInUpdatingState(cluster *management.Cluster, client *rancher.C
+				}
+			}
+		})
+	}
+}
+
+func syncRancherToAWSCheck(cluster *management.Cluster, client *rancher.Client) {
+	var err error
+	loggingTypes := []string{"api", "audit", "authenticator", "controllerManager", "scheduler"}
+	currentNodeGroupNumber := len(cluster.EKSConfig.NodeGroups)
+	initialNodeCount := *cluster.EKSConfig.NodeGroups[0].DesiredSize
+
+	By("upgrading control plane", func() {
+		syncK8sVersionUpgradeCheck(cluster, client, false)
+	})
+
+	By("scaling up the NodeGroup", func() {
+		cluster, err = helper.ScaleNodeGroup(cluster, client, initialNodeCount+1, true, true)
+		Expect(err).To(BeNil())
+
+		// Verify the existing details do NOT change in Rancher
+		for _, ng := range cluster.EKSConfig.NodeGroups {
+			Expect(*ng.Version).To(BeEquivalentTo(k8sVersion))
+			Expect(len(cluster.EKSConfig.NodeGroups)).Should(BeNumerically("==", currentNodeGroupNumber))
+		}
+
+		// Verify the new edits reflect in AWS and existing details do NOT change
+		out, err := getFromEKS("cluster", "'.[]|.Version'")
+		Expect(err).To(BeNil())
+		Expect(out).To(Equal(upgradeToVersion))
+
+		out, err = getFromEKS("nodegroup", "'.|length'")
+		Expect(err).To(BeNil())
+		Expect(strconv.Atoi(out)).To(Equal(currentNodeGroupNumber))
+
+		out, err = getFromEKS("nodegroup", "'.[]|.DesiredCapacity'")
+		Expect(err).To(BeNil())
+		Expect(strconv.ParseInt(out, 10, 64)).To(Equal(initialNodeCount + 1))
+	})
+
+	By("adding a NodeGroup", func() {
+		cluster, err = helper.AddNodeGroup(cluster, 1, client, true, true)
+		Expect(err).To(BeNil())
+
+		// Verify the existing details do NOT change in Rancher
+		Expect(*cluster.EKSConfig.KubernetesVersion).To(Equal(upgradeToVersion))
+		Expect(*cluster.EKSConfig.LoggingTypes).ShouldNot(HaveExactElements(loggingTypes))
+
+		// Verify the new edits reflect in AWS console and existing details do NOT change
+		out, err := getFromEKS("cluster", "'.[]|.Version'")
+		Expect(err).To(BeNil())
+		Expect(out).To(Equal(upgradeToVersion))
+
+		out, err = getFromEKS("nodegroup", "'.|length'")
+		Expect(err).To(BeNil())
+		Expect(strconv.Atoi(out)).To(Equal(currentNodeGroupNumber + 1))
+	})
+
+	By("Adding the LoggingTypes", func() {
+		cluster, err = helper.UpdateLogging(cluster, client, loggingTypes, true)
+		Expect(err).To(BeNil())
+
+		// Verify the existing details do NOT change in Rancher
+		Expect(*cluster.EKSConfig.KubernetesVersion).To(Equal(upgradeToVersion))
+		Expect(len(cluster.EKSConfig.NodeGroups)).To(Equal(currentNodeGroupNumber + 1))
+
+		// Verify the new edits reflect in AWS console and existing details do NOT change
+		out, err := getFromEKS("nodegroup", "'.|length'")
+		Expect(err).To(BeNil())
+		Expect(strconv.Atoi(out)).To(Equal(currentNodeGroupNumber + 1))
+
+		out, err = getFromEKS("cluster", "'.[]|.Logging|.[]|.[]|.Types'")
+		Expect(err).To(BeNil())
+		Expect(out).ShouldNot(HaveExactElements(loggingTypes))
+	})
+
+}
+
+func getFromEKS(cmd string, query string) (out string, err error) {
```

Comment:
Why not make this a helper function? It might be useful in other tests too.

Reply:
I would suggest simply returning the entire output instead of query. Query can be passed as an optional argument.

Reply:
> Returning entire output of get cluster will leak sensitive info

If the output is not printed, then how will the info leak? My suggestion was to scrub the desired data in the test instead of using json, but perhaps what you are doing here is a correct approach. In any case, this is an optional suggestion, feel free to ignore.

Reply:
No problem. 
eksctl shows output in format: `-o, --output string, specifies the output format (valid option: table, json, yaml)`
so using jq here